get_info.php

```diff
@@ -201,56 +233,110 @@ function get_list_of_surveys(
     if (isset($bind_variables[":term"])) {
         $term = true;
     }
-
-    //get list of survey_choices from a user;
+    $temp_variables = $bind_variables;
+    if($survey_id){
+        unset($temp_variables[':survey_id']);
+    }
+    //get list of surveys from a user;
     $survey_choices = get_query_result(
-        gen_query_surveys($role, $list_of_survey, $is_instance, $term, $course),
-        $bind_variables
+        gen_query_surveys($role, $term, $course, $is_instance),
+        $temp_variables
     );
-
     //get list of questions
     $list_of_quesitons = get_query_result(gen_query_questions(), []);
+    //survey result to be returned
+    $result = [];
     if (sizeof($survey_choices) <= 0) {
-        //when no data is returned
         $result = null;
-    } elseif ($list_of_survey) {
-        $result = [];
-        //specific survey data for each survey requested
-        for ($i = 0; $i < sizeof($survey_choices); $i++) {
-            $survey_data = $survey_choices[$i];
-            $survey_data["questions"] = [];
+    } elseif ($survey_id) {
+        //invalid survey_id check
+        $valid_surveys = filter_surveys($survey_choices, $bind_variables[':survey_id']);
 
-            //no number locked by course when viewing admin survey
-            if (!isset($survey_data["number_locked_by_course"])) {
-                $survey_data["number_locked_by_course"] = null;
-            }
+        //get all surveys selected
+        $survey_data = get_query_result(gen_query_get_survey_data(),[":survey_id" => $valid_surveys]);
+        $precidence = array(
+            'is_admin' => [0],
+            'is_instructor' => [1, 0],
+            'is_ta' => [2, 1, 0]
+        );
+
+        //foreach survey_data combine the choices to create one choices attribute
+        foreach ($survey_data as $index => $value) {
+            $all_choices = [];
 
-            $responses = [];
+            array_push($all_choices,$value["dept_survey_choice_id"]);
+            array_push($all_choices,$value["course_survey_choice_id"]);
+            array_push($all_choices,$value["ta_survey_choice_id"]);
+            //set role
+            $ro = '';
+
+            //Used to select question on account
+            $count = 0;
+            // this is used to keep track of null columns and determine the override from the level above
+            $num_to_set = 6;
+
+            foreach ($role as $key => $value) {
+                if ($value == "TRUE") {
```

Comment:
Is the string really `"TRUE"` in all caps?

Reply:
Yeah. Although now that I think of it, I could of kept the bit values in SQL instead of changing it to a string true/false.